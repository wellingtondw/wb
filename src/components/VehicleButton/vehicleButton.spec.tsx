import { renderWithTheme } from "../../utils/tests/helper";

import { VehicleButton } from ".";
import { fireEvent, screen } from "@testing-library/react";

describe("<VehicleButton />", () => {
  it("should be able to render correctly", () => {
    const { container } = renderWithTheme(<VehicleButton />);

    expect(screen.getByText(/carros/i)).toBeInTheDocument();
    expect(screen.getByRole("button").firstChild).toHaveClass("icon-car");

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should be able with motorcycle type", () => {
    renderWithTheme(<VehicleButton vehicleType="motorcycle" />);

    expect(screen.getByText(/motos/i)).toBeInTheDocument();
    expect(screen.getByRole("button").firstChild).toHaveClass(
      "icon-motorcycle"
    );
  });

  it("should be able to render with active status", () => {
    renderWithTheme(<VehicleButton active />);

    expect(screen.getByRole("button")).toHaveStyle({
      "border-bottom": "0.4rem solid #C02838",
    });

    expect(screen.getByText(/carros/i)).toHaveStyle({
      color: "#C02838",
    });
  });
});
