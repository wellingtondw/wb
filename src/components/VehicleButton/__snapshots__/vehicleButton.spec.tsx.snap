// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VehicleButton /> should be able to render correctly 1`] = `
<button
  class="sc-bczRLJ bwBiia sc-gsnTZi bOupwY"
>
  <span
    class="icon-car"
    style="font-size: 4.0rem;"
  />
  <div>
    <span>
      Comprar
    </span>
    <p>
      Carros
    </p>
  </div>
</button>
`;
